File: backend/internal/models/api_tokens.go
package models

import "time"

type APIToken struct {
	ID                      string     `json:"id"`
	Name                    string     `json:"name"`
	TokenPrefix             string     `json:"tokenPrefix"`
	Scopes                  []string   `json:"scopes"`
	CampaignRestrictionMode string     `json:"campaignRestrictionMode"`
	AllowedCampaignIDs      []string   `json:"allowedCampaignIds"`
	AuthenticationVersion   int        `json:"authenticationVersion"`
	LastUsedAt              *time.Time `json:"lastUsedAt,omitempty"`
	ExpiresAt               *time.Time `json:"expiresAt,omitempty"`
	RevokedAt               *time.Time `json:"revokedAt,omitempty"`
	CreatedAt               time.Time  `json:"createdAt"`
}

File: frontend/src/lib/api/apiTokens.ts
import { request } from "./request";

export type APIToken = {
  id: string;
  name: string;
  tokenPrefix: string;
  lastUsedAt?: string;
  expiresAt?: string;
  createdAt: string;
  scopes: string[];
  campaignRestrictionMode: "all" | "selected" | "legacy_all";
  allowedCampaignIds: string[];
  authenticationVersion: number;
  revokedAt?: string;
};

export type APITokenPermissions = {
  scopes: string[];
  campaignRestrictionMode: "all" | "selected";
  allowedCampaignIds: string[];
};

export const apiTokenApi = {
  apiTokens: () => request<{ tokens: APIToken[] }>("/api/auth/api-tokens"),
  createAPIToken: (name: string, expiryDays: number, permissions?: APITokenPermissions) =>
    request<{ token: APIToken; secret: string }>("/api/auth/api-tokens", {
      method: "POST",
      body: JSON.stringify({ name, expiryDays, ...permissions }),
    }),
  revokeAPIToken: (tokenId: string) =>
    request<void>(`/api/auth/api-tokens/${encodeURIComponent(tokenId)}`, {
      method: "DELETE",
    }),
};
